File: integration-test-java/src/main/java/org/seasar/doma/it/criteria/Salary.java
package org.seasar.doma.it.criteria;

import java.math.BigDecimal;
import java.util.Objects;
import org.seasar.doma.Domain;

@Domain(valueType = BigDecimal.class)
public class Salary {

  private final BigDecimal value;

  public Salary(BigDecimal value) {
    this.value = value;
  }

  public Salary(String value) {
    this(new BigDecimal(value));
  }

  public BigDecimal getValue() {
    return value;
  }

  @Override
  public String toString() {
    return value.toString();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Salary salary = (Salary) o;
    return Objects.equals(value, salary.value);
  }

  @Override
  public int hashCode() {
    return Objects.hash(value);
  }
}

File: integration-test-kotlin/src/main/kotlin/org/seasar/doma/it/criteria/Salary.kt
package org.seasar.doma.it.criteria

import org.seasar.doma.Domain
import java.math.BigDecimal

@Domain(valueType = BigDecimal::class)
class Salary(val value: BigDecimal) {

    constructor(value: String?) : this(BigDecimal(value)) {}

    override fun toString(): String {
        return value.toString()
    }
}
